src/realm/list.cpp

```diff
@@ -678,6 +690,84 @@ void Lst<Mixed>::add_index(Path& path, Index index) const
     path.emplace_back(ndx);
 }
 
+bool Lst<Mixed>::nullify(ObjLink link)
+{
+    size_t ndx = find_first(link);
+    if (ndx != realm::not_found) {
+        if (Replication* repl = Base::get_replication()) {
+            repl->list_erase(*this, ndx); // Throws
+        }
+
+        m_tree->erase(ndx);
+        return true;
+    }
+    else {
+        // There must be a link in a nested collection
+        size_t sz = size();
+        for (size_t ndx = 0; ndx < sz; ndx++) {
+            Mixed val = m_tree->get(ndx);
+            if (val.is_type(type_Dictionary)) {
+                auto dict = get_dictionary(ndx);
+                if (dict->nullify(link)) {
```

Comment:
Would it be difficult to change `nullify(link)` to become `nullify_all(link)`?  I think we only use it to clear out all references to a link, and it would be most efficient if we didn't have to start the search from the beginning each time.

Reply:
Yes, that would be an improvement, but it would require a more fundamental change to how we nullify links. Currently we handle every backilink individuallly, so if we just clear all links in the first go, we would not find any corresponding links for subsequent backlinks.

Reply:
Okay, right. This is not the place for a big change like that.